src/emulation_core/mips/line_info.rs

```diff
@@ -330,3 +333,314 @@ impl VisualDatapath for MipsDatapath {
         }
     }
 }
+
+pub fn visual_line_to_data(variable: &str, datapath_state: &UseReducerHandle<DatapathReducer>) -> LineInformation {
```

Comment:
Where do you think we should be handling the switch between MIPS and RISC-V? We're probably going to have a lot of similar variables and it might be cool to just have the UI call the same `visual_line_to_data()` function and have that function determine which value to use based on the value of `datapath_state.current_architecture`. It wouldn't require any code changes here but I'm curious what you think.

Reply:
I considered that as well but didn't know if we were keeping architecture-specific logic in their own modules. I'm open to either but I do like the compressed version of a single go-to function better. I've added it now and stuck it in `utils.rs`.